cirq/circuits/circuit_test.py

```diff
@@ -2425,3 +2425,50 @@ def test_reachable_frontier_from():
             c: 3,
             d: 5
         }
+
+
+def test_submoments():
+    a, b, c, d, e, f = cirq.LineQubit.range(6)
+    circuit = cirq.Circuit.from_ops(
+        cirq.H.on(a),
+        cirq.H.on(d),
+        cirq.CZ.on(a, d),
+        cirq.CZ.on(b, c),
+        (cirq.CNOT**0.5).on(a, d),
+        (cirq.CNOT**0.5).on(b, e),
+        (cirq.CNOT**0.5).on(c, f),
+        cirq.H.on(c),
+        cirq.H.on(e),
+    )
+
+    cirq.testing.assert_has_diagram(circuit, """
+         ┌──┐  ┌───────────────┐
+0: ───H───@─────@─────────────────────
+          │     │
+1: ───────┼@────┼────@────────────────
+          ││    │    │
+2: ───────┼@────┼────┼────@───────H───
+          │     │    │    │
+3: ───H───@─────X^0.5┼────┼───────────
+                     │    │
+4: ──────────────────X^0.5┼───────H───
+                          │
+5: ───────────────────────X^0.5───────
+         └──┘  └───────────────┘
+""")
+    cirq.testing.assert_has_diagram(circuit, """
+  0 1 2 3     4     5
+  │ │ │ │     │     │
+  H │ │ H     │     │
+┌ │ │ │ │     │     │     ┐
+│ @─┼─┼─@     │     │     │
+│ │ @─@ │     │     │     │
+└ │ │ │ │     │     │     ┘
+┌ │ │ │ │     │     │     ┐
+│ @─┼─┼─X^0.5 │     │     │
+│ │ @─┼─┼─────X^0.5 │     │
+│ │ │ @─┼─────┼─────X^0.5 │
+└ │ │ │ │     │     │     ┘
+  │ │ H │     H     │
+  │ │ │ │     │     │
+""", transpose=True)
```

Comment:
Add a test of the ascii behavior, tranposed and not transposed

Reply:
Done. I noticed that the intersections weren't `+` as I would've expected, but `|` in the regular diagram and `-` in the transpose diagram. However, this is existing behaviour not introduced by this PR (see `test_to_text_diagram_teleportation_to_diagram` for example). 

Is this a bug?

Reply:
No, that's on purpose. Subjectively, I found reading diagrams with +s harder than diagrams with |s. There was too much visual noise or something (which wasn't a problem in the unicode version because there's no whitespace breaks between the characters).